### PR TITLE
Use env_logger and log macros instead of println

### PR DESCRIPTION
**Pull request by mikeconley**
For the most part, I've just switched some println's to info! or error!. There's once case where I changed us from using a straight-up `unwrap()` to a match, which meant we could use a panic! instead of a println.

### CODE REVIEW

**Comment by mikeconley** [comment]
This is for Issue 6.

File: src/main.rs [hunk 1]
```diff
@@ -60,12 +60,12 @@ fn main() {
 
     let maybe_tus = parser::parse(&include_dirs, file_names);
 
-    if maybe_tus.is_none() {
-        println!("Specification could not be parsed.");
-        return;
-    }
-
-    let tus = maybe_tus.unwrap();
+    let tus = match maybe_tus {
```
Comment (amccreight) [inline src/main.rs]: What's the purpose of this change?

File: Cargo.lock [hunk 2]
```diff
@@ -60,6 +62,15 @@ dependencies = [
 ]
 
 [[package]]
+name = "env_logger"
+version = "0.3.5"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+dependencies = [
+ "log 0.3.6 (registry+https://github.com/rust-lang/crates.io-index)",
+ "regex 0.1.80 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (amccreight) [inline Cargo.lock]: regex increases the compile times by a fair amount, IIRC. but I guess we can worry about that later.

File: Cargo.toml [hunk 3]
```diff
@@ -10,3 +10,6 @@ lalrpop = "0.12.1"
 [dependencies]
 lalrpop-util = "0.12.1"
 getopts = "0.2.14"
+env_logger = "0.3.5"
+# We need to depend on log due to https://github.com/rust-lang/rust/issues/27812
+log = "0.3"
```
Comment (amccreight) [inline Cargo.toml]: missing newline at the end of the file

File: src/parser.rs [hunk 4]
```diff
@@ -136,7 +136,7 @@ pub fn parse_file(include_dirs: &Vec<PathBuf>, file_name: &Path) -> Result<Trans
 
 fn print_include_context(include_context: &Vec<PathBuf>) {
```
Comment (amccreight) [inline src/parser.rs]: You should probably rename this method to log_include_context.

File: src/main.rs (hunk 1 above)
Comment (amccreight) [inline src/main.rs]: Oh I see. I guess that's okay.

Comment (mikeconley) [inline src/main.rs]: Yeah, I did that mainly because of the comment [here](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap):

> In general, because this function may panic, its use is discouraged. Instead, prefer to use pattern matching and handle the None case explicitly.

File: Cargo.toml (hunk 3 above)
Comment (mikeconley) [inline Cargo.toml]: Fixed, thanks!

File: src/main.rs (hunk 1 above)
Comment (amccreight) [inline src/main.rs]: Well, thinking about this more, my initial code does not panic (because the unwrap is guarded by an is_none), while yours does. So I think the old way is better, aside from the println. Maybe revert this change, then make the println an error after that is set up?

Comment (mikeconley) [inline src/main.rs]: Yeah, that could work, and I'm happy to do it, but don't we want this case to be fatal?

File: src/parser.rs [hunk 5]
```diff
@@ -136,7 +136,7 @@ pub fn parse_file(include_dirs: &Vec<PathBuf>, file_name: &Path) -> Result<Trans
...
     for i in include_context {
-        println!("  in file included from `{}':", i.display());
+        info!("  in file included from `{}':", i.display());
```
Comment (amccreight) [inline src/parser.rs]: Oh, sorry. Thinking about this some more, these should be error instead of info (and so log_include_context doesn't really make sense as a name I guess...). We want to always print them out, because they are important for debugging the error. How does this logging work anyways? What is the default logging level? Is logging really appropriate for standard error messages? I don't know how it is supposed to be used.

Comment (mikeconley) [inline src/parser.rs]: Good questions.

According to the [env_logger documentation](http://rust-lang-nursery.github.io/log/env_logger/), `error!` is the only log level that is enabled by default - so you're right, we should probably switch this to be `error!` instead of `info!`.

That is, unless you think all of this is kind of busy-work and unnecessary. I'm using this patch series as a way of getting my feet wet in the project, but if I'm adding more of a headache than anything useful, I'm totally cool with backing off on this (I can work on the C++ generation and automated testing stuff next instead). Let me know.

File: src/main.rs (hunk 1 above)
Comment (amccreight) [inline src/main.rs]: Well, invalid input files aren't unexpected, so I think it is better to print a message (or error) and then return, rather than get the console spam from a panic.

**Comment by amccreight** [comment]
I don't really know how Rust logging is supposed to be used, so I don't
know. This printing is all just output the user wants to always see.

**Comment by mikeconley** [comment]
Alright, in that case, let's not do this right now. Maybe in the future env_logger will be more useful, but I think perhaps there's other stuff I can do that's useful here.

File: src/main.rs [hunk 6]
```diff
@@ -60,12 +63,12 @@ fn main() {
...
+        Some(m) => { m },
+        None => {
+            panic!("Specification could not be parsed.");
+        }
```
Comment (Ms2ger) [inline src/main.rs]: This code is more commonly written `let tus = maybe_tus.expect("Specification could not be parsed.");`

**Comment by amccreight** [comment]
Ah, thanks, I thought there was some way to do that.
